components/create_post/index.js

```diff
@@ -7,18 +7,40 @@ import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
 import {getCurrentChannel, getCurrentChannelStats} from 'mattermost-redux/selectors/entities/channels';
 import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
 import {get, getBool} from 'mattermost-redux/selectors/entities/preferences';
-import {makeGetMessageInHistoryItem, getPost, getMostRecentPostIdInChannel, makeGetCommentCountForPost, getLatestReplyablePostId, getCurrentUsersLatestPost} from 'mattermost-redux/selectors/entities/posts';
-import {addMessageIntoHistory, moveHistoryIndexBack, moveHistoryIndexForward, addReaction, removeReaction} from 'mattermost-redux/actions/posts';
+import {
+    getCurrentUsersLatestPost,
+    getLatestReplyablePostId,
+    getMostRecentPostIdInChannel,
+    getPost,
+    makeGetCommentCountForPost,
+    makeGetMessageInHistoryItem
+} from 'mattermost-redux/selectors/entities/posts';
+import {
+    addMessageIntoHistory,
+    moveHistoryIndexBack,
+    moveHistoryIndexForward,
+    addReaction,
+    removeReaction
+} from 'mattermost-redux/actions/posts';
 import {Posts} from 'mattermost-redux/constants';
 
-import {setEditingPost} from 'actions/post_actions.jsx';
+import {emitUserPostedEvent, postListScrollChange} from 'actions/global_actions.jsx';
+import {createPost, setEditingPost} from 'actions/post_actions.jsx';
 import {selectPostFromRightHandSideSearchByPostId} from 'actions/views/rhs';
 import {makeGetGlobalItem} from 'selectors/storage';
 import {setGlobalItem, actionOnGlobalItemsWithPrefix} from 'actions/storage';
 import {Preferences, StoragePrefixes} from 'utils/constants.jsx';
 
 import CreatePost from './create_post.jsx';
 
+function makeOnSubmitPost() {
```

Comment:
Why this is a factory and not directly a function? am I missing something?

Reply:
`post` are not immediately available. It could be moved here but it's convenient to do that in the component itself.

Reply:
Yes, but why makeOnSubmitPost function return a function, that return a function, and only called without parameters in actions binding? I think here you can have something like:
```
function onSubmitPost(post, fileInfos) {
    return () => {
        emitUserPostedEvent(post);
        createPost(post, fileInfos);
        postListScrollChange(true);
    };
}
```

And use directly that function as onSubmitPost key in the actions.

Reply:
Okay, I think yours is neat. I'll update the PR.